pkg/controllers/node/termination/terminator/terminator.go

```diff
@@ -72,24 +78,30 @@ func (t *Terminator) Taint(ctx context.Context, node *v1.Node) error {
 		if err := t.kubeClient.Patch(ctx, node, client.StrategicMergeFrom(stored)); err != nil {
 			return err
 		}
-		log.FromContext(ctx).Info("tainted node")
+		logging.FromContext(ctx).With("taint.Key", taint.Key).With("taint.Effect", taint.Effect).With("taint.Value", taint.Value).Infof("tainted node")
 	}
 	return nil
 }
 
 // Drain evicts pods from the node and returns true when all pods are evicted
 // https://kubernetes.io/docs/concepts/architecture/nodes/#graceful-node-shutdown
-func (t *Terminator) Drain(ctx context.Context, node *v1.Node) error {
+func (t *Terminator) Drain(ctx context.Context, node *v1.Node, nodeGracePeriodExpirationTime *time.Time) error {
 	pods, err := nodeutil.GetPods(ctx, t.kubeClient, node)
 	if err != nil {
 		return fmt.Errorf("listing pods on node, %w", err)
 	}
+
+	reschedulablePods := lo.Filter(pods, func(p *v1.Pod, _ int) bool { return podutil.IsReschedulable(p) })
```

Comment:
Does this capture the correct set of pods? The main thing that sticks out to me is that this excludes daemonset pods. I think what we're really trying to capture is something along the lines of `podutil.IsEvictable(p)` but without the do-not-disrupt filter, right?

Reply:
I believe it does. There's no point trying to delete a daemonset pod since it can't be relocated. We only want to delete pods that can be scheduled elsewhere. This matches the logic that karpenter uses to decide when it's safe to tear the node down.

Reply:
Isn't what you're requesting going to result in daemonset pods restarting on that node after deletion since a large number of them tolerate the disruption taint?

Reply:
I should have been more specific, we would only want to do this for daemonsets that don't tolerate our taint. You're right that many do and wouldn't be evicted. I believe this should affect the same set of pods that would be drained in our normal flow.

Reply:
Gotcha, that makes sense to me.

Reply:
Updated the logic to consider `IsWaitingEviction() && !IsTerminating()` after a DM discussion.